aioredis/util.py

```diff
@@ -319,3 +321,26 @@ def create_future(loop):
 else:
     def create_future(loop):
         return asyncio.Future(loop=loop)
+
+
+make_flat = functools.partial(functools.reduce, operator.add)
+
+
+def make_pairs(*args, **kwargs):
+    if not args and not kwargs:
+        raise TypeError("length of args or kwargs must be > 0")
+
+    elif len(args) == 1 and isinstance(args[0], dict):
+        pairs = make_flat(args[0].items())
+
+    elif len(args) % 2 != 0:
+        raise TypeError("length of args must be even number")
+
+    else:
+        pairs = args
+
+    if kwargs:
+        pairs = make_flat(kwargs.items(), pairs)
```

Comment:
What's a point of generating tuples for every iteration? Sounds like you need in:
```
pairs = list(pairs)
pairs.extend(chain.from_iterable(kwargs.items())
```

It's all said:
```
In [6]: %timeit make_flat(kwargs.items(), tuple())
100 loops, best of 3: 2.45 ms per loop

In [8]: %timeit list(chain.from_iterable(kwargs.items()))
10000 loops, best of 3: 78.2 µs per loop
```